src/Business Foundation/App/NoSeries/src/Single/NoSeriesImpl.Codeunit.al

```diff
@@ -0,0 +1,247 @@
+    begin
+        if NoSeriesCode = '' then
+            exit;
+        NoSeries.Get(NoSeriesCode);
+        if not NoSeries."Manual Nos." then
+            Error(CannotAssignManuallyErr, NoSeries.FieldCaption("Manual Nos."), NoSeries.TableCaption(), NoSeries.Code);
+    end;
+
+    procedure TestManual(NoSeriesCode: Code[20]; DocumentNo: Code[20])
+    var
+        NoSeries: Record "No. Series";
+    begin
+        if NoSeriesCode = '' then
+            exit;
+        NoSeries.Get(NoSeriesCode);
+        if not NoSeries."Manual Nos." then
+            Error(Posterr, DocumentNo);
+    end;
+
+    procedure IsManualNoSeries(NoSeriesCode: Code[20]): Boolean
+    var
+        NoSeries: Record "No. Series";
+    begin
+        if NoSeriesCode = '' then
+            exit(false);
+        if not NoSeries.Get(NoSeriesCode) then
+            exit(false);
+        exit(NoSeries."Manual Nos.");
+    end;
+
+    procedure GetLastNoUsed(var NoSeriesLine: Record "No. Series Line"): Code[20]
+    begin
+        exit(GetImplementation(NoSeriesLine).GetLastNoUsed(NoSeriesLine));
+    end;
+
+    procedure GetLastNoUsed(NoSeriesCode: Code[20]): Code[20]
+    var
+        NoSeriesLine: Record "No. Series Line";
+        NoSeriesSingle: Interface "No. Series - Single";
+    begin
+        if not GetNoSeriesLine(NoSeriesLine, NoSeriesCode, WorkDate(), true) then
+            exit('');
+
+        NoSeriesSingle := GetImplementation(NoSeriesLine);
+
+        exit(NoSeriesSingle.GetLastNoUsed(NoSeriesLine));
+    end;
+
+    procedure GetNextNo(NoSeriesCode: Code[20]; SeriesDate: Date; HideErrorsAndWarnings: Boolean): Code[20]
+    var
+        NoSeriesLine: Record "No. Series Line";
+    begin
+        NoSeriesLine."Series Code" := NoSeriesCode;
+        exit(GetNextNo(NoSeriesLine, SeriesDate, HideErrorsAndWarnings));
+    end;
+
+    procedure GetNextNo(var NoSeriesLine: Record "No. Series Line"; SeriesDate: Date; HideErrorsAndWarnings: Boolean): Code[20]
+    var
+        NoSeriesSingle: Interface "No. Series - Single";
+    begin
+        if not GetNoSeriesLine(NoSeriesLine, NoSeriesLine."Series Code", SeriesDate, HideErrorsAndWarnings) then
+            exit('');
+
+        NoSeriesSingle := GetImplementation(NoSeriesLine);
+
+        exit(NoSeriesSingle.GetNextNo(NoSeriesLine, SeriesDate, HideErrorsAndWarnings));
+    end;
+
+    local procedure GetImplementation(var NoSeriesLine: Record "No. Series Line"): Interface "No. Series - Single"
+    begin
+        if NoSeriesLine."Allow Gaps in Nos." then // TODO: Enum needs to be specified on the table and retrieved from there
+            exit(Enum::"No. Series Implementation"::Sequence);
+        exit(Enum::"No. Series Implementation"::Normal);
+    end;
+
+    procedure GetNoSeriesLine(var NoSeriesLine: Record "No. Series Line"; NoSeriesCode: Code[20]; UsageDate: Date; HideErrorsAndWarnings: Boolean): Boolean
+    var
+        NoSeries: Record "No. Series";
+    begin
+        if UsageDate = 0D then
```

Comment:
There are only two of these, not sure it makes sense to extract right now.. especially since it's only two lines, not complex logic, you need to go search in the method to figure out what is happening with usage date. Will keep in mind if we have more of these.